contracts/test/unit/PCVGuardian.sol

```diff
@@ -0,0 +1,491 @@
+        );
+        vm.stopPrank();
+
+        // move back all tokens & eth
+        vm.startPrank(addresses.governorAddress);
+        pcvGuardian.withdrawETHRatioToSafeAddress(
+            address(pcvDeposit2),
+            payable(address(pcvDeposit1)),
+            10000,
+            false,
+            false
+        );
+        pcvGuardian.withdrawERC20RatioToSafeAddress(
+            address(pcvDeposit2),
+            address(pcvDeposit1),
+            address(token),
+            10000,
+            false,
+            false
+        );
+        pcvDeposit1.deposit();
+        pcvDeposit2.deposit();
+        vm.stopPrank();
+    }
+
+    // check state changes & getters after set & unset
+    function testStateConsistency1() public {
+        address[] memory getSafeAddresses = pcvGuardian.getSafeAddresses();
+        assertEq(getSafeAddresses.length, 2);
+        assertEq(getSafeAddresses[0], address(pcvDeposit1));
+        assertEq(getSafeAddresses[1], address(pcvDeposit2));
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit1)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit2)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit3)), false);
+
+        vm.prank(addresses.governorAddress);
+        pcvGuardian.setSafeAddress(address(pcvDeposit3));
+
+        getSafeAddresses = pcvGuardian.getSafeAddresses();
+        assertEq(getSafeAddresses.length, 3);
+        assertEq(getSafeAddresses[0], address(pcvDeposit1));
+        assertEq(getSafeAddresses[1], address(pcvDeposit2));
+        assertEq(getSafeAddresses[2], address(pcvDeposit3));
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit1)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit2)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit3)), true);
+
+        vm.prank(addresses.governorAddress);
+        pcvGuardian.unsetSafeAddress(address(pcvDeposit3));
+
+        getSafeAddresses = pcvGuardian.getSafeAddresses();
+        assertEq(getSafeAddresses.length, 2);
+        assertEq(getSafeAddresses[0], address(pcvDeposit1));
+        assertEq(getSafeAddresses[1], address(pcvDeposit2));
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit1)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit2)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit3)), false);
+    }
+
+    // can't set an already safe address
+    // can't unset an already unsafe address
+    function testStateConsistency2() public {
+        address safeAddress = address(0x42);
+        vm.startPrank(addresses.governorAddress);
+
+        vm.expectRevert("PCVGuardian: not a safe address");
+        pcvGuardian.unsetSafeAddress(safeAddress);
+
+        pcvGuardian.setSafeAddress(safeAddress);
+
+        vm.expectRevert("PCVGuardian: already a safe address");
+        pcvGuardian.setSafeAddress(safeAddress);
+
+        vm.stopPrank();
+    }
+
+    // withdrawals
```

Comment:
Nit: Personal preference but I'd make this delimiter more obvious like: `///////////// WITHDRAWALS //////////`